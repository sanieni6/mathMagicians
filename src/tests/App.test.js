import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
// eslint-disable-next-line import/no-extraneous-dependencies
import TestRenderer from 'react-test-renderer';
import Calc from '../components/Calculator';

describe('Test the calc', () => {
  describe('Render the numbers', () => {
    test('It renders 0', () => {
      render(<Calc />);
      const zeros = screen.getAllByText('0');
      expect(zeros[1]).toBeInTheDocument();
    });
    test('It renders 1', () => {
      render(<Calc />);
      expect(screen.getByText('1')).toBeInTheDocument();
    });
    test('It renders 2', () => {
      render(<Calc />);
      expect(screen.getByText('2')).toBeInTheDocument();
    });
    test('It renders 3', () => {
      render(<Calc />);
      expect(screen.getByText('3')).toBeInTheDocument();
    });
    test('It renders 4', () => {
      render(<Calc />);
      expect(screen.getByText('4')).toBeInTheDocument();
    });
    test('It renders 5', () => {
      render(<Calc />);
      expect(screen.getByText('5')).toBeInTheDocument();
    });
    test('It renders 6', () => {
      render(<Calc />);
      expect(screen.getByText('6')).toBeInTheDocument();
    });
    test('It renders 7', () => {
      render(<Calc />);
      expect(screen.getByText('7')).toBeInTheDocument();
    });
    test('It renders 8', () => {
      render(<Calc />);
      expect(screen.getByText('8')).toBeInTheDocument();
    });
    test('It renders 9', () => {
      render(<Calc />);
      expect(screen.getByText('9')).toBeInTheDocument();
    });
    test('It renders AC', () => {
      render(<Calc />);
      expect(screen.getByText('AC')).toBeInTheDocument();
    });
    test('It renders +/-', () => {
      render(<Calc />);
      expect(screen.getByText('+/-')).toBeInTheDocument();
    });
    test('It renders %', () => {
      render(<Calc />);
      expect(screen.getByText('%')).toBeInTheDocument();
    });
    test('It renders รท', () => {
      render(<Calc />);
      expect(screen.getByText('รท')).toBeInTheDocument();
    });
    test('It renders x', () => {
      render(<Calc />);
      expect(screen.getByText('x')).toBeInTheDocument();
    });
    test('It renders -', () => {
      render(<Calc />);
      expect(screen.getByText('-')).toBeInTheDocument();
    });
    test('It renders +', () => {
      render(<Calc />);
      expect(screen.getByText('+')).toBeInTheDocument();
    });
    test('It renders .', () => {
      render(<Calc />);
      expect(screen.getByText('.')).toBeInTheDocument();
    });
    test('It renders =', () => {
      render(<Calc />);
      expect(screen.getByText('=')).toBeInTheDocument();
    });
  });
  describe('Test the buttons', () => {
    test('renders correctly', () => {
      const component = TestRenderer.create(<Calc />);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
    test('Test rows and cells', () => {
      let tds = 0;
      render(<Calc />);
      const rows = screen.getAllByRole('row');
      expect(rows.length).toBe(6);
      rows.forEach((row) => {
        tds = screen.getAllByRole('cell', { container: row });
      });
      expect(tds.length).toBe(20);
    });
  });
  describe('Evaluate different situations when buttons are pressed', () => {
    test('should display 15', () => {
      render(<Calc />);
      fireEvent.click(screen.getByText('1'));
      fireEvent.click(screen.getByText('5'));
      expect(document.querySelector('.result').textContent).toBe('15');
    });
    test('should display 111', () => {
      render(<Calc />);
      const button = screen.getByText('1');
      fireEvent.click(button);
      fireEvent.click(button);
      fireEvent.click(button);
      expect(document.querySelector('.result').textContent).toBe('111');
    });
    test('should display 4', () => {
      render(<Calc />);
      fireEvent.click(screen.getByText('2'));
      fireEvent.click(screen.getByText('x'));
      fireEvent.click(screen.getByText('2'));
      fireEvent.click(screen.getByText('='));
      expect(document.querySelector('.result').textContent).toBe('4');
    });
  });
});
