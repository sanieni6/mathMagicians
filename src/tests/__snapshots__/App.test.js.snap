// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the calc Test the buttons renders correctly 1`] = `
<div
  className="calc-container"
>
  <h2
    className="calc-title"
  >
    Let´s do some math!
  </h2>
  <table
    className="calc"
  >
    <tbody
      className="calc-body"
    >
      <tr
        className="firstRow row"
      >
        <td
          className="buttons result"
          colSpan={4}
        >
          <p
            className="output"
          >
            0
          </p>
        </td>
      </tr>
      <tr
        className="otherRow row"
      >
        <td
          className="buttons"
          colSpan={1}
          onClick={[Function]}
        >
          AC
        </td>
        <td
          className="buttons"
          colSpan={1}
          onClick={[Function]}
        >
          +/-
        </td>
        <td
          className="buttons"
          colSpan={1}
          onClick={[Function]}
        >
          %
        </td>
        <td
          className="buttons"
          colSpan={1}
          onClick={[Function]}
        >
          ÷
        </td>
      </tr>
      <tr
        className="otherRow row"
      >
        <td
          className="buttons"
          colSpan={1}
          onClick={[Function]}
        >
          7
        </td>
        <td
          className="buttons"
          colSpan={1}
          onClick={[Function]}
        >
          8
        </td>
        <td
          className="buttons"
          colSpan={1}
          onClick={[Function]}
        >
          9
        </td>
        <td
          className="buttons"
          colSpan={1}
          onClick={[Function]}
        >
          x
        </td>
      </tr>
      <tr
        className="otherRow row"
      >
        <td
          className="buttons"
          colSpan={1}
          onClick={[Function]}
        >
          4
        </td>
        <td
          className="buttons"
          colSpan={1}
          onClick={[Function]}
        >
          5
        </td>
        <td
          className="buttons"
          colSpan={1}
          onClick={[Function]}
        >
          6
        </td>
        <td
          className="buttons"
          colSpan={1}
          onClick={[Function]}
        >
          -
        </td>
      </tr>
      <tr
        className="otherRow row"
      >
        <td
          className="buttons"
          colSpan={1}
          onClick={[Function]}
        >
          1
        </td>
        <td
          className="buttons"
          colSpan={1}
          onClick={[Function]}
        >
          2
        </td>
        <td
          className="buttons"
          colSpan={1}
          onClick={[Function]}
        >
          3
        </td>
        <td
          className="buttons"
          colSpan={1}
          onClick={[Function]}
        >
          +
        </td>
      </tr>
      <tr
        className="otherRow row"
      >
        <td
          className="buttons"
          colSpan={2}
          onClick={[Function]}
        >
          0
        </td>
        <td
          className="buttons"
          colSpan={1}
          onClick={[Function]}
        >
          .
        </td>
        <td
          className="buttons"
          colSpan={1}
          onClick={[Function]}
        >
          =
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
